### PR TITLE
ENH: add as_scipy_operator, closes #323

### LINKED ISSUE #323: Interfacing with scipy's Linear algebra
It seems scipy has a  [LinearOperator](http://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.sparse.linalg.LinearOperator.html) abstraction that allows access to their rather extensive [solvers](http://docs.scipy.org/doc/scipy/reference/sparse.linalg.html).

We could easily add support for this by adding a helper function

```python
def as_flat_array(arr):
    if hasattr(arr, 'order'):
        return arr.asarray().ravel(arr.order)
    else:
        return arr.asarray()

def as_scipy_operator(op):
    assert op.is_linear
    
    shape = (op.range.size, op.domain.size)
    
    def matvec(v):
        return as_flat_array(op(v))
        
    def rmatvec(v):
        return as_flat_array(op.adjoint(v))
        
    dtype = op.domain.dtype
    assert op.range.dtype == dtype
    
    return scipy.sparse.linalg.LinearOperator(shape=shape,
                                              matvec=matvec,
                                              rmatvec=rmatvec,
                                              dtype=dtype)
```

### PR DESCRIPTION
**Pull request by adler-j**
Very simple code that adds interface to scipys linear solvers, also has a simple example.

### CODE REVIEW

File: examples/solvers/scipy_solvers.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Example of interfacing with scipy solvers
+
+This example solves the problem
+
+    -Laplacian(x) = b
```
Comment (kohr-h) [inline examples/solvers/scipy_solvers.py]: Backticks?

File: examples/solvers/scipy_solvers.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Example of interfacing with scipy solvers
```
Comment (kohr-h) [inline examples/solvers/scipy_solvers.py]: No space in the beginning, dot at the end.

File: examples/solvers/scipy_solvers.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+This example solves the problem
+
+    -Laplacian(x) = b
+
+Where b is a dirac delta impulse
```
Comment (kohr-h) [inline examples/solvers/scipy_solvers.py]: small "w", dot at the end

File: examples/solvers/scipy_solvers.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import odl
+
+# Create discrete space, a square from [-1, 1] x [-1, 1] with (11 x 11) points
+space = odl.uniform_discr([-1, -1], [1, 1], [11, 11])
+
+# Create odl operator for negative laplacian
+laplacian = -odl.Laplacian(space)
+
+# Create right hand side, a gaussian around the point (0, 0)
+rhs = space.element(lambda x: np.exp(-(x[0]**2 + x[1]**2)/0.1**2))
```
Comment (kohr-h) [inline examples/solvers/scipy_solvers.py]: Can you PEP-8ify this?

File: examples/solvers/scipy_solvers.py (hunk 3 above)
Comment (kohr-h) [inline examples/solvers/scipy_solvers.py]: Not in sync with the code, as far as I see it.

File: odl/operator/oputils.py [hunk 5]
```diff
@@ -178,6 +180,69 @@ def power_method_opnorm(op, niter, xstart=None):
...
+    This is intended to be used with the scipy sparse linear solvers.
+
+    Parameters
+    ----------
+    op : `Operator`
+        A linear operator that should be wrapped
+
+    Returns
+    -------
+    `scipy.sparse.linalg.LinearOperator` : linear_op
```
Comment (kohr-h) [inline odl/operator/oputils.py]: If this link works, you could use single backticks in the one-line summary, too.

**Comment by kohr-h** [comment]
Cool stuff. I guess we don't have to bother with the simple iterative methods any longer. We should definitely have a documentation site on this, otherwise it's hard to find for users.

**Comment by adler-j** [comment]
> Cool stuff. I guess we don't have to bother with the simple iterative methods any longer.

Slightly disagree here since these work very badly with cuda and distributed systems.

> We should definitely have a documentation site on this, otherwise it's hard to find for users.

We should have a whole page on "numpy / scipy" interactions on site, it should cover this and more.

**Comment by kohr-h** [comment]
> Slightly disagree here since these work very badly with cuda and distributed systems.

Yes, sure. And product spaces won't work easily either.

**Comment by kohr-h** [comment]
Looking good

**Pull request merged**